### ISSUE TITLE
Use the dependency injection system

### ISSUE THREAD

**Issue by MFAshby** [IBM/FHIR#2687]
**Is your feature request related to a problem? Please describe.**
I'd like to implement the extension proposed in https://github.com/IBM/FHIR/issues/2680, but I'm finding it difficult due to many static methods calling `Instant.now()`. This appears to be a wider problem in the FHIR server codebase, it's not easy to change the wiring of dependent services.

**Describe the solution you'd like**
Avoid using static methods except when they are actually pure functions (i.e. no side effects, and they depend only on the passed in parameters). Use the dependency injection system to wire together classes and their dependencies. CDI should be used in this case, as provided by [openliberty server](https://openliberty.io/guides/cdi-intro.html).

This proposal might also allow removal of boilerplate code like `com.ibm.fhir.audit.AuditLogServiceFactory` which appears to be doing manual dependency injection.

**Describe alternatives you've considered**
One possible alternative in this case is to simply have some `static` factory for the current date and time, which can be mutated if necessary.

**Acceptance Criteria**

1.  
   GIVEN [a precondition]
    AND [another precondition]
   WHEN [test step]
    AND [test step]
   THEN [verification step]
    AND [verification step]

**Additional context**

**Comment by lmsurpre** [comment]
Yes, I've actually had this thought for a while, so I'm glad you've opened this enhancement request.  I'm not sure when we will get to it, but if you're interested in working with us on using more of CDI we're definitely open to the idea.
